crypto/http/http_client.c

```diff
@@ -463,6 +463,21 @@ static int check_set_resp_len(OSSL_HTTP_REQ_CTX *rctx, size_t len)
     return 1;
 }
 
+static int may_still_retry(time_t max_time, int *ptimeout)
+{
+    time_t time_diff;
+
+    if (max_time != 0) {
+        if (max_time < time(NULL)) {
```

Comment:
What about calling `time(NULL)` just once not twice?  On some machines it's free but on others it can be expensive.  It also avoids a race condition where the time changes.

Reply:
Good points! Done.